v-next/hardhat-utils/src/eth.ts

```diff
@@ -13,6 +17,18 @@ export function isAddress(value: unknown): boolean {
   return typeof value === "string" && /^0x[0-9a-f]{40}$/i.test(value);
 }
 
+/**
+ * Checks if a value is an Ethereum address and if the checksum is valid.
+ *
+ * @param value The value to check.
+ * @returns True if the value is an Ethereum address with a valid checksum, false otherwise.
+ */
+export async function isValidChecksumAddress(value: unknown): Promise<boolean> {
```

Comment:
This is nitpicky, but this could be sync. I'd also be ok with this remaining async.